src/operations/execute_operation.ts

```diff
@@ -118,6 +118,12 @@ export async function executeOperation<
     );
   }
 
+  timeoutContext ??= TimeoutContext.create({
```

Comment:
Shouldn't we be constructing the timeout with the `options` from the operation?  Otherwise, `timeoutMS` and other options won't be properly inherited from the client -> db -> collection -> operation.

Reply:
timeoutMS is the only option here that we get from the operation. `waitQueueTimeoutMS` and `serverSelectionTimeoutMS` are strictly client options that can't be set at the database, collection or operation level